frontend/src/app/workflows/workflows.component.ts

```diff
@@ -112,19 +109,20 @@ export class WorkflowsComponent implements OnInit {
     );
   }
 
-  setIndexFromIframe(index: number, diagramType: 'pages' | 'objects' | 'class') {
+  @HostListener('window:message', ['$event'])
+  handleGlobalMessages(event: MessageEvent) {
+    const messageData = JSON.parse(event.data);
```

Comment:
event.data does not need to be manually encoded with JSON, you can just pass an object.

Reply:
right about that